### DISCUSSION TITLE
PUF 2 step reaction FDS

### DISCUSSION THREAD

**Discussion by bjmali3389** [firemodels/fds#12917] [General]
Hi
I am trying to model the burning behaviour of three polyurethane foams placed side by side. I would like to model PUF pyrolysis as two step reaction involving decomposition into TDI and polyol which ultimates generates soot.  However I seem to be encountering problem in running the code saying material is incorrectly described. Could anyone kindly point out what mistakes I might be making and how to rectify? I am using simple chemistry. Do i need to specify two reactions like in complex stoichiometry?
[~urethane_block_fire4_96G.txt](https://github.com/firemodels/fds/files/15285409/urethane_block_fire4_96G.txt)

**Comment by Er9y714** [comment]
Have you checked the User guide for 2 step chemistry setup? You will also find example fds files in example folder (explained in User guide). Further examples can be found in Verification guide as well.

**Reply by bjmali3389** [reply]
Thank you for your response. Yes, I checked the user guide examples for 2 step chemistry. However, I have a small doubt with regard to it. In the examples for propane flame, it is a two step reaction involving CO conversion to CO, both of which are predfined species.  However, in my case, if I want to model as PUF-->TDI+Polyol, Polyol--> soot+gas.  Do I define polyol as a  new species with properties or define only as a material.? And as for reaction, do I define two reactions or a single reaction for PUF and then in foam pyrolysis, i add polyol as a byproduct and then polyol pyrolysis gives soot as byproduct. Sorry for the confusion.

**Comment by mcgratta** [comment]
The input parameter `THRESHOLD_TEMPERATURE` is no longer used. Update your input file and version of FDS.

**Reply by bjmali3389** [reply]
Thank you for the information.
